### ISSUE TITLE
Handling Vue2 reactivity caveats for useFind/useGet

### ISSUE THREAD

**Issue by miguelrk** [marshallswain/feathers-pinia#65]
**TLDR:** UI is blank (fails to populate with useFind/useGet data) due to reactivity issues when using `vue2` with `feathers-pinia`.

### Context
Basically when I refresh the page, all find()s and get()s are executed successfully (as shown by the devtools network tab), but they won't be rendered in the UI components (despite stores containing the fetched data). Only for example changing page, and then returning to the previous page re-flushes the UI and renders the components with the already fetched data (without refetch).

This I'm sure has to do with the vue2 reactivity caveats. After much thought and time spent trying to fix this, I figured these issues are inevitable for `vue2` due to the Observer-bases reactivity (compared to the newer Proxy based reactivity of `vue3` which fixes many of these issues).

### Workaround for Vue2
For users stuck with `vue2` for now, I settled with using a simple workaround by wrapping the `useFind` and `useGet` composables as follows:

```typescript
import type { BaseModel, ServiceStore, UseFindOptionsStandalone, UseGetOptionsStandalone } from 'feathers-pinia'
import { useFind, useGet } from 'feathers-pinia'

interface M extends BaseModel {}

export function useFindVue2(
  store: ServiceStore,
  options: UseFindOptionsStandalone<M>,
) {
  const { items: _items, haveLoaded, ...rest } = useFind(options)

  const items = computed(() => haveLoaded.value
    ? _items.value
    : store.findInStore(options.params.value).data,
  )

  return { items, haveLoaded, ...rest }
}

export function useGetVue2(
  store: ServiceStore,
  options: UseGetOptionsStandalone<M>,
) {
  const { item: _item, hasLoaded, ...rest } = useGet(options)

  // NOTE: check if _item.value is truthy to force computed to re-evaluate ternary
  // operator since _item.value is sometimes null just after hasLoaded.value is set
  // to true, so watch both to trigger updates when _item finally updates (if it does)
  const item = computed(() => hasLoaded.value && _item.value // check both
    ? _item.value
    : store.getFromStore(options.id?.value ?? null, options.params?.value) ?? {},
  )

  return { item, hasLoaded, ...rest }
}
```

I've been using the above and it is working properly, however, there might be certain edge cases where this does not solve the reactivity issues, though I don't believe this will break.

@marshallswain [mentioned](https://discord.com/channels/509848480760725514/930351539804065842/1018340469127782400) conditionally returning these as `useFind`/`useGet` for `vue2` apps might be something worth adding to `feathers-pinia`, so opening up this issue.

**Comment by marshallswain** [comment]
@miguelrk will you please let me know if you continue to need this workaround in order for Vue 2 to work with `useFind`?  Thank you!

**Comment by miguelrk** [comment]
Sure @marshallswain ! Haven't finished migrating to the new `useFind`/`useGet` yet, but I'll be sure to report back when I am 👍🏼

**Comment by miguelrk** [comment]
**TLDR:** no luck, it appears to break even more after upgrading to `feathers-pinia@v2`

### Context

None of the following works for `feathers-pinia@^2.0.0-pre.13` on `vue2` when using pinia setup stores and `ModelFn`s:

```ts
const Model = useSomeModel()
const store = useSomeStore()

// With the legacy useFindWatched:
const { items } = store.useFindWatched({ params })
const { items } = store.useFindWatched({ model: Model, params })
const { items } = store.useFindWatched({ model: store.Model, params })
const { items } = Model.useFindWatched({ params })
const { items } = Model.useFindWatched({ model: Model, params })
const { items } = Model.useFindWatched({ model: store.Model, params })
// With the new useFind():
const { data: items } = store.useFindWatched(params)
const { data: items } = Model.useFindWatched(params)
// With the aforementioned workaround useFindVue2:
const { items } = useFindVue2(store, { model: store.Model, params })
const { items } = useFindVue2(store, { model: Model, params })
// With the aforementioned workaround useFindModelVue2:
const { items } = useFindVue2(Model, { params })
const { items } = useFindVue2(Model, { model: Model, params })
```

**`items` will remain `[]` for all of the above (also in pinia devtools)**, despite it being a valid `Ref` and the network tab indeed making the network requests and returning non-empty data. At this point I'm starting to think it might be an issue of `vue-demi` or `@vue/reactivity` having conflicting implementations for `vue2`/`vue3` 😕

**Comment by miguelrk** [comment]
**UPDATE:** the reactive implementation differs between vue2 and vue3 and is what is apparently causing issues

I believe I found the root of the issue...

I forked the `feathers-pinia-vite` template from Marshall (so `vue3`) to test a single service with my API and it worked (incredibly fast btw)... 

**I noticed however that the `data` returned by `Model.useFind()` is a `Watcher2` in vue2, but but a `ComputedRefImpl` in vue3 (see image) :warning: **

```ts
const query = computed(() => ({}))
const { data } = Model.useFind({ query })
console.log(data) // <--------- IMPORTANT: 'data' will be different for vue2 and 
```
**vue3:**
![Screenshot 2023-01-19 11 57 43 PM](https://user-images.githubusercontent.com/18708856/213583373-6e700871-ae3e-4734-8d13-d747a22085ff.png)

**vue2:**
![Screenshot 2023-01-20 12 06 06 AM](https://user-images.githubusercontent.com/18708856/213583363-d25f530a-cfe2-439a-a603-8b7bdf602c78.png)

**Comment by miguelrk** [comment]
**UPDATE:** I figured out what was causing the reactivity issue in vue2 (but also in vue3). Note this does not have anything to do with the vue2 reactivity caveats, those are still there, so the above workaround using `useFindVue2`/`useGetVue2` might still be required in vue2 (I've updated to vue3 now, and there, it's working flawlessly)

### From [this thread](https://discord.com/channels/509848480760725514/1067361773805637652/1067902879832211466):

I wanted to let you know I found the cause of this issue!  :pray: :tada: :beers:  I'm happy to say it simply a minor mistake, not a bug... 

I did refactor my entire app, file by file, to ensure the import race-conditions weren't the problem, it appears they weren't.

What was actually the issue, was calling `syncWithStorage(store)` with a `pinia` store instead of a `feathers-pinia` store (docs must be updated https://v2.feathers-pinia.pages.dev/guide/storage-sync.html#sync-with-storage, will submit a PR shortly). Also, registration of `syncWithStorage` must happen in the model file e.g. `src/models/task.ts` and not within the `useTaskStore` definition in `src/stores/service.tasks.ts`. 

#### In summary:

:one: **WHAT DIDN'T WORK: registering pinia store in `src/stores/service.tasks.ts`:**
```ts
export const useTaskStore = () => {
  ...
  const store = useStore(pinia)
  connectModel(name, useTaskModel, () => store)
  onModelReady(name, () => {
    // NOTE pinia store
    syncWithStorage(store, ['ids', 'itemsById'])
  })
  return store
}
```

:two: **WHAT ALSO DIDN'T WORK: registering feathers-pinia store in `src/stores/service.tasks.ts`:**
```ts
export const useTaskStore = () => {
  ...
  const store = useStore(pinia)
  connectModel(name, useTaskModel, () => store)
  onModelReady(name, () => {
    // NOTE feathers-pinia store (can't be called here despite in callback)
    syncWithStorage(useTaskStore(), ['ids', 'itemsById'])
  })
  return store
}
```

:three: **WHAT DOES WORK: registering feathers-pinia store in `src/models/task.ts`:**
```ts
export const useTaskModel = () => {
  ...
  onModelReady(name, () => {
    service.hooks({ around: { all: [...feathersPiniaHooks(Model)] } })
    // NOTE feathers-pinia store
    syncWithStorage(useTaskStore(), ['ids', 'itemsById'])
  })
  connectModel(name, () => Model, useTaskStore)
  return Model
}
```

This works also with auto-imports, so that wasn't an issue.

**Comment by marshallswain** [comment]
Thanks for figuring this out. It's a big relief. 😁
